### DISCUSSION TITLE
Is Ortur using grblHAL?

### DISCUSSION THREAD

**Discussion by phil-barrett** [grblHAL/core#163] [General]
I saw a Hackaday post about [Ortur going open source](https://hackaday.com/2022/07/18/ortur-laser-will-go-open-source/#comments).  In the comments there was this:

> The bootup banner on one of the machines says grblHAL

Is this common knowledge and I missed it?

**Comment by terjeio** [comment]
I did not know about it but I suspected grblHAL would eventually be used commercially due to its portability/open architecture.

Merging code for [new plugins](https://github.com/grblHAL/plugins#i-have-written-a-plugin-and-i-want-to-make-it-available-to-grblhal-users) and/or drivers is something I am reluctant to do as this could lead to providing support for that code, I'll rather link to the repos instead. Pull requests for changes to the core and plugins will be accepted if beneficial.

**Comment by troth530** [comment]
@phil-barrett  and @terjeio yes they are. They enabled UART on 2209s that I'd like to see worked back into grblHal mainstream.

**Reply by terjeio** [reply]
Ortur said they would publish the modified source, but the repo they created was deleted shortly after. I do not know if they have created a new one later on.

grblHAL has support for TMC2209 drivers provided that the MCU/board combo implements the low level interface required.

**Reply by troth530** [reply]
This covers all but the Laser Master 3 which is the first one with UART Trinamic working.  If the pinmap they used on this version of firmware is any indication of what they used for Laser Master 3, they did not use addressing.

https://github.com/Ortur-Tech/Ortur_Laser_Firmware_17X-18X

**Reply by terjeio** [reply]
Their TMC2209 UART code is found [here](https://github.com/Ortur-Tech/Ortur_Laser_Firmware_17X-18X/blob/4849ac85fc1f1a15c47d0dbbfc3e69450922ad32/ESPLaserMasterGrbl/main/trinamic/tmc_interface.c#L59-L76) and [here](https://github.com/Ortur-Tech/Ortur_Laser_Firmware_17X-18X/blob/4849ac85fc1f1a15c47d0dbbfc3e69450922ad32/ESPLaserMasterGrbl/main/single_uart.c).
